From: Taehee Yoo <ap420073@gmail.com>
Date: Sun, 8 Mar 2020 01:19:30 +0000
Subject: bareudp: remove unnecessary udp_encap_enable() in
 bareudp_socket_create()
Patch-mainline: v5.7-rc1
Git-commit: 2baecda37f4ef4414be15452a333fe4fd13c0df3
References: jsc#SLE-15172

In the current code, udp_encap_enable() is called in
bareudp_socket_create().
But, setup_udp_tunnel_sock() internally calls udp_encap_enable().
So, udp_encap_enable() is unnecessary.

Signed-off-by: Taehee Yoo <ap420073@gmail.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Thomas Bogendoerfer <tbogendoerfer@suse.de>
---
 drivers/net/bareudp.c |    3 ---
 1 file changed, 3 deletions(-)

--- a/drivers/net/bareudp.c
+++ b/drivers/net/bareudp.c
@@ -250,9 +250,6 @@ static int bareudp_socket_create(struct
 	tunnel_cfg.encap_destroy = NULL;
 	setup_udp_tunnel_sock(bareudp->net, sock, &tunnel_cfg);
 
-	if (sock->sk->sk_family == AF_INET6)
-		udp_encap_enable();
-
 	rcu_assign_pointer(bareudp->sock, sock);
 	return 0;
 }
